tensorflow/python/kernel_tests/xent_op_test.py

```diff
@@ -195,18 +195,18 @@ def testSecondGradient(self):
                                                    name="xent")
       loss = math_ops.reduce_sum(x)
 
-      trivia_loss = -math_ops.reduce_sum(l * math_ops.log(nn_ops.softmax(f)))
+      hessians = gradients_impl.hessians(loss, [f])[0]
 
-      hessians = sess.run(gradients_impl.hessians(loss, [f]))
-      trivia_hessians = sess.run(gradients_impl.hessians(trivia_loss, [f]))
-
-      self.assertAllClose(hessians, trivia_hessians)
+      err = gradient_checker.compute_gradient_error(f, [12], hessians, [12, 12])
```

Comment:
Ah, right: it wouldn't have failed but it also wouldn't have checked the right thing.  It would have checked that the gradient of the Hessian was consistent.   Your correct way is indeed correct.